fix(elasticsearch): advancedOptions in domain has no effect

jumic opened the pull request:
Property `advancedOptions` in ElasticSearch domain did have no effect because the assignment was missing.
* add assignment for advancedOptions to fix issue
* test cases
* describe function in readme

Fixes #14067

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: 9e730114d003e52210a5b6152fa974206c7dce08
* Result: SUCCEEDED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=4850a862-b0e9-44e4-85b9-db56e2736ae5%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

peterwoodworth reviewed (dismissed):
This PR looks great! Thank you for fixing this oversight

BenChaimberg commented on packages/@aws-cdk/aws-elasticsearch/test/domain.test.ts:
```diff
@@ -1645,6 +1645,34 @@ describe('unsigned basic auth', () => {
   });
 });
 
+describe('advanded options', () => {
```
```suggestion
describe('advanced options', () => {
```

BenChaimberg commented on packages/@aws-cdk/aws-elasticsearch/README.md:
```diff
@@ -290,3 +290,16 @@ new Domain(stack, 'Domain', {
+new Domain(stack, 'Domain', {
+    version: ElasticsearchVersion.V7_7,
+    advancedOptions: {
+        'rest.action.multi.allow_explicit_index': 'false',
```
let's put all three documented options here

jumic replied:
done

BenChaimberg reviewed (changes requested):
Can you please also add one or more of these to the integ.elasticsearch.ts integration test?

jumic commented:
I added the integration test as requested and changed the readme.

I also hat to change the policy in the integration test. I received the message 'Apply a restrictive access policy to your domain' when I deployed the stack. My changes where not active at this time. I replaced `iam.AnyPrincipal()`by `iam.AccountRootPrincipal()` to use a valid policy.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: 56bd14528ffb4c2161682e060aa416f54bf0f013
* Result: FAILED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=2d48dcfa-7d14-4164-abb7-1f09e29cb670%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: 248087dd91db4399cf3bbe5eaa89c1a2cb6a42ec
* Result: FAILED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=e4f7c268-ee94-4298-8048-03f39f82cfa7%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: 9179d959de6c12d579f7b9e1a43d8c2e8eb71cd7
* Result: SUCCEEDED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=5637454c-0de7-437f-aa78-255c11d30689%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->
